doc/types.md

```diff
@@ -0,0 +1,59 @@
+# Types
+
+Every programming language needs a foundation in sound type theory.

```

Comment:
This is rather ***strong*** statement considering there are many widely used programming languages without types. And even those with types are rarely concerned about the soundness of their type theory. For example, Haskell's type theory is well known to be unsound, but it is fine with that.

Reply:
A precise statement might be "every theorem prover" or "strongly types programming language with integrated correctness proofs".